GoogleDataTransport/GDTCORLibrary/GDTCORFlatFileStorage.m

```diff
@@ -354,6 +356,76 @@ - (void)removeLibraryDataForKey:(nonnull NSString *)key
   });
 }
 
+- (BOOL)hasEventsForTarget:(GDTCORTarget)target {
+  NSString *searchPath = [GDTCORFlatFileStorage pathForTarget:target qosTier:nil mappingID:nil];
+  NSFileManager *fm = [NSFileManager defaultManager];
+  NSDirectoryEnumerator *enumerator = [fm enumeratorAtPath:searchPath];
+  while ([enumerator nextObject]) {
+    if ([enumerator.fileAttributes[NSFileType] isEqual:NSFileTypeRegular]) {
+      return YES;
+    }
+  }
+  return NO;
+}
+
+- (nullable id<GDTCORStorageEventIterator>)iteratorWithSelector:
```

Comment:
It would be nice to know why we need the iterator. Would enumeration using a block (e.g. like [NSArray](https://developer.apple.com/documentation/foundation/nsarray/1415846-enumerateobjectsusingblock?language=objc)) work better? Is there a way to avoid  using `dispatch_sync`?

Reply:
Mostly because it could be costly (in terms of memory and CPU) to block a thread while a set of events is recreated from storage. Doing it one by one means that an event can be recreated and destroyed one at a time.

Reply:
I should've asked my 3 questions separately to get answers to all of them 😄.